### ISSUE TITLE
IndexOutOfRangeException on decoding base64 bytes with LF characters included (>= .NET 5 only)

### ISSUE THREAD

**Issue by paatrofimov** [dotnet/runtime#76918]
### Description

I have upgraded runtime from netcoreapp3.1 to NET 6 and the code which was transforming base64 bytes containing LF characters to string without errors started to crash with IndexOutOfRangeException. Behavior is the same for Windows and Linux platforms.

### Reproduction Steps

```
[TestFixture]
public class Fixture
{
    [Test]
    public void Crashes_on_runtime_greater_or_equal_to_NET_5()
    {
        var txt = "YWJj\nZGVm"; // abc\ndef
        var base64Bytes = Encoding.UTF8.GetBytes(txt);
        var stream = new MemoryStream(base64Bytes);
        var base64Transform = new FromBase64Transform();
        var cryptoStream = new CryptoStream(stream, base64Transform, CryptoStreamMode.Read);

        var result = new MemoryStream();
        cryptoStream.CopyTo(result);

        Console.WriteLine(Encoding.UTF8.GetString(result.ToArray()));
    }
}
```

### Expected behavior

"abcdef" in the output, which is happening for example on netcoreapp3.1 runtime.

### Actual behavior

```
System.IndexOutOfRangeException : Index was outside the bounds of the array.
   at System.Security.Cryptography.FromBase64Transform.TransformFinalBlock(Byte[] inputBuffer, Int32 inputOffset, Int32 inputCount)
   at System.Security.Cryptography.CryptoStream.ReadAsyncCore(Memory`1 buffer, CancellationToken cancellationToken, Boolean useAsync)
   at System.Security.Cryptography.CryptoStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Security.Cryptography.CryptoStream.CopyTo(Stream destination, Int32 bufferSize)
   at System.IO.Stream.CopyTo(Stream destination)
   at ClassLibrary1.Fixture.Crashes_on_runtime_greater_or_equal_to_NET_5() in E:\cm_1\drive\ClassLibrary1\Class1.cs:line 20
```

### Regression?

Regression happened after the upgrade from netcoreapp3.1 to NET 5/NET 6.

**Comment by paatrofimov** [comment]
It is worth mentioning that the same input but using a CRLF delimiter works without exceptions.
The problem only occurs with the LF delimiter.

**Comment by gfoidl** [comment]
The crash happens in https://github.com/dotnet/runtime/blob/4850c88171a9970890ecf3d5fca1c0c7258b465e/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Base64Transforms.cs#L267 when not enough bytes are in `tmpBuffer` which is to happen with `\n` in your example (len=1), whilst for `\r\n` len=2 (or any other whitespace instead of `\r`).
So the problem is in `FromBase64Transform` (test-hole?).

Side note: `Base64.DecodeFromUtf8` returns `InvalidData` (consumed=4, written=3) as expected and documented.
And for completeness cf. https://github.com/dotnet/runtime/issues/76020#issuecomment-1274968258

**Comment by paatrofimov** [comment]
FYI guys from SO suggested padding base64 bytes to the length of % 4 as a workaround until this is fixed. It seems to work and the implementation is simpler than adding a middleware stream which would remove all whitespaces from base64 bytes input.

**Comment by CodeCasterNL** [comment]
@gfoidl we arrived to the same conclusion, see https://stackoverflow.com/questions/74039771/indexoutofrangeexception-on-decoding-base64-bytes-with-lf-characters-included/74041328#74041328.
